### ISSUE TITLE
Adding tests

### ISSUE THREAD

**Issue by girotobial** [Ortho4XP/Ortho4XP#5]
To help with refactoring work, and to support more confidence in the code when accepting changes, I think it would be wise to setup unit tests. 

This is obviously a large body of work, and the existing code could be quite difficult to test. Therefore i propose we try to add tests to code as we make changes. Hopefully with time we will start building up some coverage.

**Comment by artesim** [comment]
Yes, definitely : I had tried using pytest on a private branch at some point, but if you know a better one, I could learn from your experience

**Comment by girotobial** [comment]
Yeah I think pytest is the way to go. I've opened a PR #8 to add it to the dev dependencies.
